app/routes/__index/certificate/index.tsx

```diff
@@ -1,43 +1,68 @@
 import { Flex, Center } from '@chakra-ui/react';
-import type { Certificate } from '@prisma/client';
-import { useState } from 'react';
+import type { LoaderArgs, ActionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
+import { useRevalidator } from '@remix-run/react';
+import { useInterval } from 'react-use';
 
-import undrawSvg from '~/assets/undraw_processing_re_tbdu.svg';
+import { requireUser } from '~/session.server';
+import pendingSvg from '~/assets/undraw_processing_re_tbdu.svg';
 import Loading from '~/components/display-page';
 import CertificateAvailable from '~/components/certificate/certificate-available';
 import CertificateRequestView from '~/components/certificate/certificate-request';
 import { useUser } from '~/utils';
+import { getCertificateByUsername } from '~/models/certificate.server';
+import { addCertRequest } from '~/queues/certificate/certificate-flow.server';
 
-export default function CertificateIndexRoute() {
-  const user = useUser();
-  const [certificateRequested, setCertificateRequested] = useState(false);
-  const [loading, setLoading] = useState(false);
+export const loader = async ({ request }: LoaderArgs) => {
+  const user = await requireUser(request);
 
-  const currentDate: Date = new Date(2023, 2);
+  const certificate = (await getCertificateByUsername(user.username)) || {};
```

Comment:
`typedjson` prevents me from moving forward if it receives an empty array... I can  pass an empty object `return typedjson(certificate || {});` or throw an error, but if I do the latter nothing will ever be executed

Reply:
Probably if you do this, it will be OK:

```ts
return typedjson({ certificate });
```

So you always return an Object, but it may or may not have a `certificate` inside.

Reply:
It's causing issue with `certificate?.status` not existing for `certificate`

Reply:
oh nvm it's merely due to the change in the structure of the object